saleor/events/models/OrderEvent.py

```diff
@@ -0,0 +1,274 @@
+
+        if source is not None and source.is_anonymous:
+            source = None
+
+        return cls(
+            order=order, type=OrderEvents.EMAIL_SENT.value,
+            user=source,
+            parameters={
+                'email': order.get_user_current_email(),
+                'email_type': email_type.value})
+
+    @classmethod
+    def email_resent_event(
+            cls, *,
+            order: Order, email_type: OrderEventsEmails,
+            source: User) -> models.Model:
+        raise NotImplementedError
+
+    @classmethod
+    def draft_created_event(
+            cls, *, order: Order, source: User) -> models.Model:
+        return cls(
+            order=order, type=OrderEvents.DRAFT_CREATED.value, user=source)
+
+    @classmethod
+    def draft_added_products(
+            cls, *,
+            order: Order, source: User,
+            order_lines: List[Tuple[int, ProductVariant]]) -> models.Model:
+
+        return cls(
+            order=order, type=OrderEvents.DRAFT_ADDED_PRODUCTS.value,
+            user=source,
+            parameters={
+                'lines': cls._lines_per_quantity_to_str_line_list(order_lines)
+            })
+
+    @classmethod
+    def draft_removed_products(
+            cls, *,
+            order: Order, source: User,
+            order_lines: List[Tuple[int, ProductVariant]]
+    ) -> models.Model:
+
+        return cls(
+            order=order, type=OrderEvents.DRAFT_REMOVED_PRODUCTS.value,
+            user=source,
+            parameters={
+                'lines': cls._lines_per_quantity_to_str_line_list(
+                    order_lines)})
+
+    @classmethod
+    def placed_event(
+            cls, order: Order, source: User, from_draft=False) -> models.Model:
+        event_type = (
+            OrderEvents.PLACED_FROM_DRAFT if from_draft else OrderEvents.PLACED
+        )
+
+        if source.is_anonymous:
+            source = None
+
+        return cls(
+            order=order, type=event_type.value, user=source)
+
+    @classmethod
+    def draft_oversold_items_event(
+            cls, *,
+            order: Order, source: User,
+            oversold_items: List[str]) -> models.Model:
+        return cls(
+            order=order, type=OrderEvents.OVERSOLD_ITEMS.value,
+            user=source,
+            parameters={
+                'oversold_items': oversold_items})
+
+    @classmethod
+    def cancelled_event(
+            cls, *,
+            order: Order, source: User) -> models.Model:
+        return cls(
+            order=order, type=OrderEvents.CANCELED.value,
+            user=source)
+
+    @classmethod
+    def manually_marked_as_paid_event(
+            cls, *,
+            order: Order, source: User) -> models.Model:
+        return cls(
+            order=order, type=OrderEvents.ORDER_MARKED_AS_PAID.value,
+            user=source)
+
+    @classmethod
+    def fully_paid_event_event(cls, *, order: Order) -> models.Model:
```

Comment:
The word `event` is repeated here.

Would it make sense if we had a convention of naming events like `model_action_event`? For payment events below we actually use it: "payment_captured_event, "payment_refunded_event" and for draft-related actions "draft_something_event" etc. Here we would then have "order_fully_paid_event",  "order_marked_as_paid_event", "order_email_sent" etc. What do you think?

Reply:
Great suggestion. I refactored that and put a note about the naming convention in the docs.